plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/stock-filter/frontend.ts

```diff
@@ -1,100 +1,50 @@
+		toggleFilter: () => {
+			const productFiltersContext = getContext< ProductFiltersContext >(
+				'woocommerce/product-filters'
 			);
-
-			const filters = checkboxContext.items
-				.filter( ( item ) => {
-					return item.checked;
-				} )
-				.map( ( item ) => {
-					return item.value;
-				} );
-
-			navigate( getUrl( filters.join( ',' ) ) );
-		},
-		onDropdownChange: () => {
-			const dropdownContext = getContext< DropdownContext >(
-				'woocommerce/interactivity-dropdown'
-			);
-
-			const selectedItems = dropdownContext.selectedItems;
-			const items = selectedItems || [];
-			const filters = items.map( ( i ) => i.value );
-
-			navigate( getUrl( filters.join( ',' ) ) );
-		},
-		updateProducts: ( event: HTMLElementEvent< HTMLInputElement > ) => {
-			// get the active filters from the url:
-			const url = new URL( window.location.href );
 			const currentFilters =
-				url.searchParams.get( 'filter_stock_status' ) || '';
+				productFiltersContext.params[ filterStockStatusKey ];
 
 			// split out the active filters into an array.
 			const filtersArr =
-				currentFilters === '' ? [] : currentFilters.split( ',' );
+				currentFilters === undefined ? [] : currentFilters.split( ',' );
 
-			// if checked and not already in activeFilters, add to activeFilters
-			// if not checked and in activeFilters, remove from activeFilters.
-			if ( event.target.checked ) {
-				if ( ! currentFilters.includes( event.target.value ) ) {
-					filtersArr.push( event.target.value );
-				}
-			} else {
-				const index = filtersArr.indexOf( event.target.value );
-				if ( index > -1 ) {
-					filtersArr.splice( index, 1 );
-				}
-			}
+			const { ref } = getElement();
+			const value = ref.getAttribute( 'value' );
 
-			navigate( getUrl( filtersArr.join( ',' ) ) );
-		},
-		removeFilter: () => {
-			const { value } = getContext< { value: string } >();
-			// get the active filters from the url:
-			const url = new URL( window.location.href );
-			const currentFilters =
-				url.searchParams.get( 'filter_stock_status' ) || '';
-
-			// split out the active filters into an array.
-			const filtersArr =
-				currentFilters === '' ? [] : currentFilters.split( ',' );
-
-			const index = filtersArr.indexOf( value );
+			const newFilterArr = filtersArr.includes( value )
+				? [ ...filtersArr.filter( ( filter ) => filter !== value ) ]
+				: [ ...filtersArr, value ];
 
-			if ( index > -1 ) {
-				filtersArr.splice( index, 1 );
-			}
+			productFiltersContext.params = {
+				...productFiltersContext.params,
+				[ filterStockStatusKey ]: newFilterArr.join( ',' ),
+			};
+		},
+		clearFilters: () => {
+			const productFiltersContext = getContext< ProductFiltersContext >(
+				'woocommerce/product-filters'
+			);
+			const updatedParams = productFiltersContext.params;
 
-			navigate( getUrl( filtersArr.join( ',' ) ) );
+			productFiltersContext.params = {
+				...updatedParams,
+				[ filterStockStatusKey ]: '',
```

Comment:
When you clear the filter, the URL becomes this: `http://woo.test/shop/?filter_stock_status=`. We probably need to update the `woocommerce/product-filters` store to handle null params, as it's easier to clear the params with a null value compared to manually deleting the param by key like I did [here](https://github.com/woocommerce/woocommerce/blob/a7e601181661f4808d0810a26142388bd24b7899/plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/attribute-filter/frontend.ts#L72-L83).

Reply:
Thanks for catching it, it should be fixed now 🙇